src/huggingface_hub/hf_api.py

```diff
@@ -3685,9 +3687,9 @@ def update_repo_settings(
         if repo_type is None:
             repo_type = constants.REPO_TYPE_MODEL  # default repo type
 
-        # Check if both gated and private are None
-        if gated is None and private is None:
-            raise ValueError("At least one of 'gated' or 'private' must be provided.")
+        # Check if gated, private, and xet_enabled are None
+        if gated is None and private is None and xet_enabled is None:
```

Comment:
if condition gets bigger we could move this check below and do a 

```py
if len(payload) == 0:
    raise ValueError("At least one setting must be updated.")
```
=> after that, no need to update it if we add more settings